src/client/method-unit-actions.c

```diff
@@ -0,0 +1,388 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#include "method-unit-actions.h"
+#include "client.h"
+
+int method_lifecycle_action_on(Client *client, char *node_name, char *unit, char *method) {
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *message = NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *job_result = NULL;
+        char *job_path = NULL, *result = NULL, *node = NULL;
+        uint32_t id = 0;
+        int r = 0;
+
+        r = assemble_object_path_string(NODE_OBJECT_PATH_PREFIX, node_name, &client->object_path);
+        if (r < 0) {
+                return r;
+        }
+
+        r = sd_bus_match_signal(
+                        client->api_bus,
+                        NULL,
+                        BC_INTERFACE_BASE_NAME,
+                        BC_MANAGER_OBJECT_PATH,
+                        MANAGER_INTERFACE,
+                        "JobRemoved",
+                        match_job_removed_signal,
+                        client);
+
+        if (r < 0) {
+                fprintf(stderr, "Failed to match signal\n");
+                return r;
+        }
+
+        r = sd_bus_call_method(
+                        client->api_bus,
+                        BC_INTERFACE_BASE_NAME,
+                        client->object_path,
+                        NODE_INTERFACE,
+                        method,
+                        &error,
+                        &message,
+                        "ss",
+                        unit,
+                        "replace");
+        if (r < 0) {
+                fprintf(stderr, "Failed to issue method call: %s\n", error.message);
+                return r;
+        }
+
+        r = sd_bus_message_read(message, "o", &job_path);
+        if (r < 0) {
+                fprintf(stderr, "Failed to parse response message: %s\n", strerror(-r));
+                return r;
+        }
+
+        job_result = wait_for_job(client, job_path);
+        if (job_result == NULL) {
+                return -EIO;
+        }
+
+        r = sd_bus_message_read(job_result, "uosss", &id, &job_path, &node, &unit, &result);
+        if (r < 0) {
+                fprintf(stderr, "Can't parse job result\n");
+                return r;
+        }
+
+        printf("Unit %s %s operation result: %s\n", unit, method, result);
+
+        return r;
+}
+
+int create_message_new_method_call(
```

Comment:
Should this method be `static`?

Reply:
This one actually was moved by mistake, it goes to client.c, and it's a part of the Client interface.
Also renamed it and `wait_for_job` to `client_*` for it to be clear